[Boundary Docs] Adds targets and static host catalog usage docs (#1698)
hashicorp/web-unified-docs: content/boundary/v0.21.x/content/docs/targets/create/tcp/web.mdx

- You have created an org and a project where you want to set up the web target. This example calls the org `dev-projects` and the project `web-infrastructure`. The example may refer to the org by ID (such as `o_c2Ah7bPMj5`) or the project by ID (such as `p_UXVxIgpZ8U`), depending on whether you follow the UI, CLI, or Terraform workflows.

This example also assumes that you want to set up an access model with the following permissions:

- Unauthenticated users cannot access the address range
- Authenticated but unauthorized users cannot view the web target
- Authenticated and authorized users can start sessions to web target

This access model allows users connected to the web target through Boundary to access the address range, but they cannot connect to the target unless you grant a user access to it.

With these prerequisites in place, you can set up the permissions and targets to restrict access to the web target to an authenticated user. This example grants access to a single user, but using a group for authenticated users is more realistic.

<Tabs>
<Tab heading="Admin Console" group="admin">

### Configure users and grants

> 

Author: I restructured this page with an additional set of tab groups. This allowed me to reduce some redundant text and get rid of the empty headers.

---
layout: docs
page_title: Create a web TCP target
description: >-
  Learn how to create a web TCP target.
---

# Create a web TCP target

TCP targets provide a generic way to connect to any service Boundary has access to, including HTTP and HTTPS services.

The following examples use a direct target address for simplicity, but HashiCorp recommends that you [configure host catalogs and host sets](/boundary/docs/hosts) for scaled production deployments.

Complete the following steps to create a TCP target to connect to a web service. This example assumes an HTTPS service.

<Tabs>
<Tab heading="Admin Console" group="admin">

1. Log in to Boundary.
1. Select an org, and then select the project where you want to create a target.

> 1. Select **Targets** under *Project Actions*.

Reviewer: ```suggestion
1. Select **Targets** under **Project Actions**.
```

}

resource "boundary_alias_target" "test.example.com" {
  name                      = "test-example-web-server"
  description               = "Target alias for test-example-web-server"
  scope_id                  = "global"
  value                     = "test.example.com"
  destination_id            = boundary_target.example.id
}
```

</Tab>
</Tabs>

## Configure a brokered application credential

@include 'configuration-reference/credentials/configure-brokered-credentials.mdx'

You can configure credentials for the TCP target using:

- Static credentials (`username_password` or `ssh_private_key`)
- Vault generic credential library

Refer to the [Configure targets with credential brokering](/boundary/docs/credentials/configure-credential-brokering) page to learn how to configure a target with credential brokering.

## Example

The following scenario assumes:

- A web server is deployed in your private network (Nginx, Apache, IIS, etc).
- [A Boundary worker](/boundary/docs/workers) is deployed in your network and registered with the Boundary control plane.
- Your application domain has a registered DNS A-record and is resolvable by authenticated users on your network.

> - SSL signed certificates are installed on your web server, and you have distributed the root certificate to clients you want to connect to the web target. Clients should add the cert to their keychain to trust the CA that generated it.

Reviewer: ```suggestion
- SSL-signed certificates are installed on your web server, and you have distributed the root certificate to clients you want to connect to the web target. Clients should add the cert to their keychain to trust the CA that generated it.
```

- Unauthenticated users cannot access the address range
- Authenticated but unauthorized users cannot view the web target
- Authenticated and authorized users can start sessions to web target

This access model allows users connected to the web target through Boundary to access the address range, but they cannot connect to the target unless you grant a user access to it.

With these prerequisites in place, you can set up the permissions and targets to restrict access to the web target to an authenticated user. This example grants access to a single user, but using a group for authenticated users is more realistic.

### Configure users and grants

To model the different access protocols, you should create an authorized and unauthorized user. These represent users that should and should not have access to the web application hosted on the target.

<Tabs>
<Tab heading="Admin Console" group="admin">

#### Create the unauthorized and authorized users:

1. Log in to Boundary.
1. Make sure you are in the `global` scope. The left sidebar will confirm you are at the **Global Level**.
1. Select **Users** under **Global IAM**.
1. Create an unauthorized user:
  1. Click **New User**.

>   1. Complete the following fields:

Reviewer: These numbered steps are not showing up as indented, but I'm not sure why, they look OK to me. Do they need one more space of indentation?

     - **Name**: (Required) A name for identification purposes.
     The name must be unique. This example calls the user `unauthorized`.
     - **Description**: (Optional) An optional description of the target for identification purposes.
1. Create an authorized user:
  1. Navigate back to the **Users** page and click **New User**.
  1. Complete the following fields:
     - **Name**: (Required) A name for identification purposes.
     The name must be unique. This example calls the user `authorized`.
     - **Description**: (Optional) An optional description of the target for identification purposes.

>      Click **Save**.

Reviewer: ```suggestion
1. Create an authorized user:
   1. Navigate back to the **Users** page and click **New User**.
   1. Complete the following fields:
      - **Name**: (Required) A name for identification purposes.
      The name must be unique. This example calls the user `authorized`.
      - **Description**: (Optional) An optional description of the target for identification purposes.
   1. Click **Save**.
```

1. Add another grant in the **New Grant** field: `type=user;actions=list-resolvable-aliases;ids=*`, then click **Add**.
1. Click **Save**.

There are two grant strings in the example above:

`"type=auth-token;ids=*;actions=read:self"` This grant ensures that the user can authenticate to the Client Agent, if you are using a [transparent sessions workflow](/boundary/docs/targets/connections).
`"type=user;actions=list-resolvable-aliases;ids=*"` This grant ensures that the user can read the aliases in a transparent sessions workflow.

#### Assign read permissions to the authorized user in the `global` scope:

1. Make sure you are in the `global` scope.
1. Click **Roles** in the sidebar. Click **New Role**.
1. Complete the following fields:
   - **Name**: (Required) A name for identification purposes.
   The name must be unique. This example calls the role `authorized_global_role`.
   - **Description**: (Optional) An optional description of the target for identification purposes.
   Click **Save**.
1. Select the **Principals** tab for the new role. Click **Add Principals**.

> 1. Check the box next to the `authorized` user in the `global` scope. Click **Add Principals**.

Reviewer: ```suggestion
1. Select the box next to the `authorized` user in the `global` scope and click **Add Principals**.
```

1. Select the **Grants** tab for the role.
1. In the **New Grant** field, enter the grant string: `ids=*;type=*;actions=read`, then click **Add**.
1. Add another grant in the **New Grant** field: `type=auth-token;ids=*;actions=read:self`, then click **Add**.
1. Add another grant in the **New Grant** field: `type=user;actions=list-resolvable-aliases;ids=*`, then click **Add**.
1. Click **Save**.

The grant string `"ids=*;type=*;actions=read"` ensures that the user can read all resources in the `global` scope.

#### Assign full access permissions for the authorized user in the org scope:

1. Click on **Back to Global** on the upper-right part of the page.
1. On the **Orgs** page, select the name of your org (such as `dev-projects` or the ID like `o_c2Ah7bPMj5`).
1. Click **Roles** in the sidebar. Click **New Role**.
1. Complete the following fields:
   - **Name**: (Required) A name for identification purposes.
   The name must be unique. This example calls the role `authorized_org_role`.
   - **Description**: (Optional) An optional description of the target for identification purposes.
   Click **Save**.
1. Select the **Principals** tab for the new role. Click **Add Principals**.

> 1. Check the box next to the `authorized` user in the `global` scope. Click **Add Principals**.

Reviewer: ```suggestion
1. Select the box next to the `authorized` user in the `global` scope and click **Add Principals**.
```

1. Select the **Grants** tab for the role.
1. In the **New Grant** field, enter the grant string: `ids=*;type=*;actions=*`, then click **Add**.
1. Click **Save**.

The grant string `"ids=*;type=*;actions=*"` ensures that the user can perform all actions in the scope.

#### Assign full access permissions for the authorized user in the project scope:

> 1. Select **Projects** in the left-hand sidebar under *Org Level*.

Reviewer: ```suggestion
1. Select **Projects** in the left-hand sidebar under **Org Level**.
```
